### PR TITLE
[Bugfix][Connector-V2][maxcompute] spark engine support date type 

### LINKED ISSUE #3869: The maxcompute source does not support the conversion of date type
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

The maxcompute source does not support the conversion of date type ,when batch job running 。

### SeaTunnel Version

v2.3.0

### SeaTunnel Config

```conf
env {
  spark.app.name = "seatunnel"
  spark.executor.instances = 2
  spark.executor.cores = 1
  spark.executor.memory = "1g"
  // 这个配置必需填写
   #spark.sql.catalogImplementation = "hive"
}
source {
  Maxcompute {
    accessId="accessId"
    accesskey="accesskey"
    endpoint="http://service.cn-shanghai.maxcompute.aliyun.com/api"
    project="noah_ods"
    table_name="bak_go_goal_new_t_index_info"
    #partition_spec="<your partition spec>"
    #split_row = 10000
  }
}


sink {
  Console {}
}
```

### Running Command

```shell
./bin/start-seatunnel-spark-connector-v2.sh --config config/maxcompute2rds1.conf -e client -m 'local[2]'
```

### Error Exception

```log
"ERROR ParallelBatchPartitionReader: BatchPartitionReader execute failed.
org.apache.seatunnel.connectors.seatunnel.maxcompute.exception.MaxcomputeConnectorException: ErrorCode:[COMMON-12], ErrorDescription:[Source reader operation failed, such as (open, close) etc...] - java.lang.ClassCastException: java.util.Date cannot be cast to java.lang.Long
  at org.apache.seatunnel.connectors.seatunnel.maxcompute.util.MaxcomputeTypeMapper.resolveObject(MaxcomputeTypeMapper.java:285)"
```

### Flink or Spark Version

 spark-2.4.8

### Java or Scala Version

scala-2.13.10

### Screenshots

_No response_

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by msftgitsklsd**
## Purpose of this pull request

close https://github.com/apache/incubator-seatunnel/issues/3869

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-common/src/main/java/org/apache/seatunnel/translation/spark/common/serialization/InternalRowConverter.java [hunk 1]
```diff
@@ -76,7 +77,11 @@ private static Object convert(Object field, SeaTunnelDataType<?> dataType) {
                 SeaTunnelRowType rowType = (SeaTunnelRowType) dataType;
                 return convert(seaTunnelRow, rowType);
             case DATE:
-                return (int) ((LocalDate) field).toEpochDay();
+                if (field instanceof Date ){
```
Comment (hailin0) [inline seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-common/src/main/java/org/apache/seatunnel/translation/spark/common/serialization/InternalRowConverter.java]: revert other all unrelated changes

Comment (msftgitsklsd) [reply seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-common/src/main/java/org/apache/seatunnel/translation/spark/common/serialization/InternalRowConverter.java]: done

**Comment by hailin0** [comment]
check ci error

https://github.com/apache/incubator-seatunnel/actions/runs/4051453143/jobs/6975834349

**Comment by stdnt-xiao** [comment]
> check ci error
> 
> https://github.com/apache/incubator-seatunnel/actions/runs/4051453143/jobs/6975834349

Why doesn't anyone help to review this pr
https://github.com/apache/incubator-seatunnel/pull/3894

**Comment by msftgitsklsd** [comment]
> check ci error
> 
> https://github.com/apache/incubator-seatunnel/actions/runs/4051453143/jobs/6975834349

fixed the checkstyle

**Review by hailin0 — COMMENTED** [review]
update org.apache.seatunnel.connectors.seatunnel.maxcompute.util.MaxcomputeTypeMapper#resolveObject

```java
switch (fieldType.getSqlType()) {
case DATE:
     return ((Date) field).toInstant()
         .atZone(ZoneId.systemDefault())
         .toLocalDate()
         .toEpochDay();
```

@maruko-code 
revert other update

https://github.com/apache/incubator-seatunnel/pull/2052

**Comment by liugddx** [comment]
Please resolve code conflicts.

**Comment by msftgitsklsd** [comment]
> update org.apache.seatunnel.connectors.seatunnel.maxcompute.util.MaxcomputeTypeMapper#resolveObject
> 
> ```java
> switch (fieldType.getSqlType()) {
> case DATE:
>      return ((Date) field).toInstant()
>          .atZone(ZoneId.systemDefault())
>          .toLocalDate()
>          .toEpochDay();
> ```
> 
> @maruko-code revert other update
> 
> #2052

thanks, already updated the code and tested

**Comment by hailin0** [comment]
> update org.apache.seatunnel.connectors.seatunnel.maxcompute.util.MaxcomputeTypeMapper#resolveObject
> 
> ```java
> switch (fieldType.getSqlType()) {
> case DATE:
>      return ((Date) field).toInstant()
>          .atZone(ZoneId.systemDefault())
>          .toLocalDate()
>          .toEpochDay();
> ```
> 
> @maruko-code revert other update
> 
> #2052

spark handles date as int

https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-2.4/src/main/java/org/apache/seatunnel/translation/spark/serialization/InternalRowConverter.java#L189

**Comment by msftgitsklsd** [comment]
> > update org.apache.seatunnel.connectors.seatunnel.maxcompute.util.MaxcomputeTypeMapper#resolveObject
> > ```java
> > switch (fieldType.getSqlType()) {
> > case DATE:
> >      return ((Date) field).toInstant()
> >          .atZone(ZoneId.systemDefault())
> >          .toLocalDate()
> >          .toEpochDay();
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @maruko-code revert other update
> > #2052
> 
> spark handles date as int
> 
> https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-translation/seatunnel-translation-spark/seatunnel-translation-spark-2.4/src/main/java/org/apache/seatunnel/translation/spark/serialization/InternalRowConverter.java#L189

sorry I am not geting the point, the connector-maxcompute can read the source which has date type filed is not enough?
I sync data from maxconpute to mysql (the table has two date fields) and get expected result

**Comment by hailin0** [comment]
root exception cause

```java
"ERROR ParallelBatchPartitionReader: BatchPartitionReader execute failed.
org.apache.seatunnel.connectors.seatunnel.maxcompute.exception.MaxcomputeConnectorException: ErrorCode:[COMMON-12], ErrorDescription:[Source reader operation failed, such as (open, close) etc...] - java.lang.ClassCastException: java.util.Date cannot be cast to java.lang.Long
  at org.apache.seatunnel.connectors.seatunnel.maxcompute.util.MaxcomputeTypeMapper.resolveObject(MaxcomputeTypeMapper.java:285)"
```
